qiskit/algorithms/eigensolvers/numpy_eigensolver.py

```diff
@@ -0,0 +1,291 @@
+                    eigval, eigvec = np.linalg.eigh(operator.to_matrix())
+                else:
+                    eigval, eigvec = np.linalg.eig(operator.to_matrix())
+            else:
+                if operator.is_hermitian():
+                    eigval, eigvec = scisparse.linalg.eigsh(sp_mat, k=self._k, which="SA")
+                else:
+                    eigval, eigvec = scisparse.linalg.eigs(sp_mat, k=self._k, which="SR")
+            indices = np.argsort(eigval)[: self._k]
+            eigval = eigval[indices]
+            eigvec = eigvec[:, indices]
+        self._ret.eigenvalues = eigval
+        self._ret.eigenstates = eigvec.T
+
+    def _get_ground_state_energy(self, operator: BaseOperator | PauliSumOp) -> None:
+        if self._ret.eigenvalues is None or self._ret.eigenstates is None:
+            self._solve(operator)
+
+    def _get_energies(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None,
+    ) -> None:
+        if self._ret.eigenvalues is None or self._ret.eigenstates is None:
+            self._solve(operator)
+
+        if aux_operators is not None:
+            aux_op_vals = []
+            for i in range(self._k):
+                aux_op_vals.append(
+                    self._eval_aux_operators(aux_operators, self._ret.eigenstates[i])
+                )
+            self._ret.aux_operator_eigenvalues = aux_op_vals
+
+    @staticmethod
+    def _eval_aux_operators(
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp],
+        wavefn: np.ndarray,
+        threshold: float = 1e-12,
+    ) -> ListOrDict[tuple[complex, complex]]:
+
+        values: ListOrDict[tuple[complex, complex]]
+
+        # As a list, aux_operators can contain None operators for which None values are returned.
+        # As a dict, the None operators in aux_operators have been dropped in compute_eigenvalues.
+        if isinstance(aux_operators, list):
+            values = [None] * len(aux_operators)
+            key_op_iterator = enumerate(aux_operators)
+        else:
+            values = {}
+            key_op_iterator = aux_operators.items()
+        for key, operator in key_op_iterator:
+            if operator is None:
+                continue
+            value = 0.0
+            if operator.coeff != 0:
+                mat = operator.to_spmatrix()
+                # Terra doesn't support sparse yet, so do the matmul directly if so
+                # This is necessary for the particle_hole and other chemistry tests because the
+                # pauli conversions are 2^12th large and will OOM error if not sparse.
+                if isinstance(mat, scisparse.spmatrix):
+                    value = mat.dot(wavefn).dot(np.conj(wavefn))
+                else:
+                    # TODO: How can we remove this opflow dependency?
+                    value = StateFn(operator, is_measurement=True).eval(wavefn)
```

Comment:
I am not familiar enough with opflow/quantum-info to know how to replace this evaluation properly, given that the wavefunction is expressed as an array/matrix, but I think that we should try to remove as many opflow dependencies as possible. Does anybody have any suggestion?

Reply:
I believe you solved this. FYI in Aqua before opflow this was used to compute the value https://github.com/Qiskit/qiskit-aqua/blob/f7528590f1a702c11240f450294893e7f1125cd7/qiskit/aqua/operators/matrix_operator.py#L262-L273 you can see it called from here https://github.com/Qiskit/qiskit-aqua/blob/f7528590f1a702c11240f450294893e7f1125cd7/qiskit/aqua/algorithms/classical/exact_eigen_solver/exact_eigen_solver.py#L127